File: Java/Array/LongestWord.java
package Array;

public class LongestWord {

    String getLongestWord(String str) {
        str = str.replaceAll("[^0-9a-zA-Z ]", "");
        String arr[] = str.split(" ");
        int max = 0, index = 0;
        for (int i = 0; i < arr.length; i++) {
            int len = arr[i].length();
            if (len > max) {
                max = len;
                index = i;
            }
        }
        return arr[index];
    }

    public static void main(String[] args) {
        LongestWord longestWord = new LongestWord();
        System.out.println(longestWord.getLongestWord("fun&!! time"));
    }

}

File: JS/Array/LongestWord.js
/*
Longest Word
Have the function LongestWord(sen) take the sen parameter being passed and return the longest word in the string. If there are two or more words that are the same length, return the first word from the string with that length. Ignore punctuation and assume sen will not be empty. Words may also contain numbers, for example "Hello world123 567"
*/

const longestWord = (str) => {
  str = str.replace(/[^a-z0-9A-Z ]/gi, "");
  let max = 0;
  let index = 0;
  str = str.split(" ");
  str.forEach((val, i) => {
    if (val.length > max) {
      max = val.length;
      index = i;
    }
    return val;
  });
  return str[index];
};

console.log(longestWord("hello!! world"));
